src/main/java/com/coreos/jetcd/internal/impl/ClientConnectionManager.java

```diff
@@ -126,6 +127,17 @@ ExecutorService getExecutorService() {
     return configureStub(supplier.apply(getChannel()), getToken());
   }
 
+  /**
+   * create and add a new token to channel's head.
+   *
+   * @param supplier the stub supplier
+   * @param <T> the type of stub
+   * @return the attached stub
+   */
+  <T extends AbstractStub<T>> T newStubWithNewToken(Function<ManagedChannel, T> supplier) {
+    return configureStub(supplier.apply(getChannel()), generateToken(getChannel()));
```

Comment:
I know it is hard code as 5 min as ttl expiration if etcd uses the default auth mechanism. If I remember the auth code correctly,  any req for the such get() would renew the token TTL internally. However, the issue with renewing token is that we have to do that every 5 min. If there is a long live client that doesn't do anything when initialized, then renew the token every 5 min seems excessive. If we just refresh the token when needed, that might be simpler and save bandwidth.